storagenode/console/service.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package console
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/version"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/bandwidth"
+	"storj.io/storj/storagenode/pieces"
+)
+
+// DB exposes methods for managing SNO dashboard related data.
+type DB interface {
+	GetSatelliteIDs(ctx context.Context, from, to time.Time) (storj.NodeIDList, error)
+}
+
+// Service is handling storage node operator related logic
+type Service struct {
+	log *zap.Logger
+
+	consoleDB   DB
+	bandwidthDB bandwidth.DB
+	pieceInfoDB pieces.DB
+	kademlia    *kademlia.Kademlia
+	version     *version.Service
+
+	allocatedBandwidth memory.Size
+	allocatedDiskSpace memory.Size
+	walletAddress      string
+	startedAt          time.Time
+	versionInfo        version.Info
+}
+
+// NewService returns new instance of Service
+func NewService(log *zap.Logger, consoleDB DB, bandwidth bandwidth.DB, pieceInfo pieces.DB, kademlia *kademlia.Kademlia, version *version.Service,
+	allocatedBandwidth, allocatedDiskSpace memory.Size, walletAddress string, versionInfo version.Info) (*Service, error) {
+	if log == nil {
+		return nil, errs.New("log can't be nil")
+	}
+
+	if consoleDB == nil {
+		return nil, errs.New("consoleDB can't be nil")
+	}
+
+	if bandwidth == nil {
+		return nil, errs.New("bandwidth can't be nil")
+	}
+
+	if pieceInfo == nil {
+		return nil, errs.New("pieceInfo can't be nil")
+	}
+
+	if version == nil {
+		return nil, errs.New("version can't be nil")
+	}
+
+	if kademlia == nil {
```

Comment:
This could be condensed into a single check since the information that something is missing should be sufficient to figure out what is the problem.